src/poetry/core/factory.py

```diff
@@ -409,6 +409,20 @@ def validate(
                             'Use "allow-prereleases" instead.'
                         )
 
+                if "extras" in config:
+                    for extra_name, requirements in config["extras"].items():
+                        extra_name = canonicalize_name(extra_name)
+
+                        for req in requirements:
+                            for dependency in config["dependencies"].keys():
+                                if req == dependency:
```

Comment:
I think it's necessary to `canonicalize_name()` requirements as well as dependencies.